tests/integration/cqlengine/query/test_queryset.py

```diff
@@ -218,16 +218,47 @@ def test_queryset_with_distinct(self):
         query3 = TestModel.objects.distinct(['test_id', 'attempt_id'])
         self.assertEqual(len(query3._distinct_fields), 2)
 
-    def test_defining_only_and_defer_fails(self):
+    def test_defining_only_fields(self):
         """
-        Tests that trying to add fields to either only or defer, or doing so more than once fails
+        Tests defining only fields
         """
 
-    def test_defining_only_or_defer_on_nonexistant_fields_fails(self):
+        q = TestModel.objects.only(['attempt_id', 'description'])
+        self.assertEqual(q._select_fields(), ['attempt_id', 'description'])
+
+        with self.assertRaises(query.QueryException):
+            TestModel.objects.only(['nonexistant_field'])
+
+        with self.assertRaises(query.QueryException):
+            TestModel.objects.only(['description']).only(['attempt_id'])
+
+        q = TestModel.objects.only(['attempt_id', 'description'])
+        q = q.defer(['description'])
+        self.assertEqual(q._select_fields(), ['attempt_id'])
+
+        # PYTHON-560
+        q = TestModel.objects.filter(test_id=0).only(['test_id', 'attempt_id', 'description'])
+        self.assertEqual(q._select_fields(), ['attempt_id', 'description'])
+
+    def test_defining_defer_fields(self):
         """
-        Tests that setting only or defer fields that don't exist raises an exception
+        Tests defining defer fields
         """
 
+        q = TestModel.objects.defer(['attempt_id', 'description'])
+        self.assertEqual(q._select_fields(), ['test_id', 'expected_result', 'test_result'])
+
+        with self.assertRaises(query.QueryException):
+            TestModel.objects.defer(['nonexistant_field'])
```

Comment:
Oups. Thanks
On Jun 3, 2016 12:42 PM, "Adam Holmberg" <notifications@github.com> wrote:

> In tests/integration/cqlengine/query/test_queryset.py
> <https://github.com/datastax/python-driver/pull/593#discussion_r65738659>:
>
> >          """
> >
> > +        q = TestModel.objects.defer(['attempt_id', 'description'])
> > +        self.assertEqual(q._select_fields(), ['test_id', 'expected_result', 'test_result'])
> > +
> > +        with self.assertRaises(query.QueryException):
> > +            TestModel.objects.defer(['nonexistant_field'])
>
> same typo
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/datastax/python-driver/pull/593/files/86d4f180abc80dc5fc13650cb7a3c2f6dfb073ab#r65738659>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAgfhFS_lRMe-B0ixpAlAK-FOxaIxAinks5qIFlogaJpZM4ItfvA>
> .
>